sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/TransactionalBatch.java

```diff
@@ -0,0 +1,351 @@
+ *
+ * TransactionalBatch batch = TransactionalBatch.createTransactionalBatch(new Cosmos.PartitionKey(activityType));
+ * batch.createItem<ToDoActivity>(test1);
+ * batch.replaceItem<ToDoActivity>(test2.id, test2);
+ * batch.upsertItem<ToDoActivity>(test3);
+ * batch.deleteItem("reading");
+ *
+ * TransactionalBatchResponse response = container.executeTransactionalBatch(batch);
+ *
+ * if (!response.isSuccessStatusCode()) {
+ *      // Handle and log exception
+ *      return;
+ * }
+ *
+ * // Look up interested results - e.g., via typed access on operation results
+ *
+ * TransactionalBatchOperationResult result = response.get(0);
+ * ToDoActivity readActivity = result.getItem(ToDoActivity.class);
+ *
+ * }</pre>
+ *
+ * <b>Example</b>
+ * <p>This example atomically reads a set of items as a batch.
+ * <pre>{@code
+ * String activityType = "personal";
+ *
+ * TransactionalBatch batch = TransactionalBatch.createTransactionalBatch(new Cosmos.PartitionKey(activityType));
+ * batch.readItem("playing");
+ * batch.readItem("walking");
+ * batch.readItem("jogging");
+ * batch.readItem("running")v
+ *
+ * TransactionalBatchResponse response = container.executeTransactionalBatch(batch);
+ * List<ToDoActivity> resultItems = new ArrayList<ToDoActivity>();
+ *
+ * for (int i = 0; i < response.size(); i++) {
+ *     TransactionalBatchOperationResult result = response.get(0);
+ *     resultItems.add(result.getItem(ToDoActivity.class));
+ * }
+ *
+ * }</pre>
+ * <p>
+ * <b>See:</b>
+ * <a href="https://docs.microsoft.com/azure/cosmos-db/concepts-limits">Limits on TransactionalBatch requests</a>.
+ */
+@Beta(Beta.SinceVersion.V4_7_0)
+public final class TransactionalBatch {
+
+    private final List<ItemBatchOperation<?>> operations;
+    private final PartitionKey partitionKey;
+
+    TransactionalBatch(PartitionKey partitionKey) {
+        checkNotNull(partitionKey, "expected non-null partitionKey");
+
+        this.operations = new ArrayList<>();
+        this.partitionKey = partitionKey;
+    }
+
+    /**
+     * Initializes a new instance of {@link TransactionalBatch}
+     * that will contain operations to be performed across multiple items in the container with the provided partition
+     * key in a transactional manner
+     *
+     * @param partitionKey the partition key for all items in the batch.
+     *
+     * @return A new instance of {@link TransactionalBatch}.
+     */
+    public static TransactionalBatch createTransactionalBatch(PartitionKey partitionKey) {
+        return new TransactionalBatch(partitionKey);
+    }
+
+    /**
+     * Adds an operation to create an item into the batch.
+     *
+     * @param item A JSON serializable object that must contain an id property.
+     * @param <T> The type of item to be created.
+     *
+     * @return The transactional batch instance with the operation added.
+     */
+    public <T> CosmosItemOperation createItem(T item) {
```

Comment:
NIT - as discussed I think naming the factory methods xxxItemOperation here - like createItemOperation or upsertItemOperation would be clearer.